deltachat-jsonrpc/src/api.rs

```diff
@@ -1829,6 +1829,17 @@ impl CommandApi {
         WebxdcMessageInfo::get_for_message(&ctx, MsgId::new(instance_msg_id)).await
     }
 
+    /// Get href from a webxdc message
+    async fn get_webxdc_href(
+        &self,
+        account_id: u32,
+        instance_msg_id: u32,
+    ) -> Result<Option<String>> {
+        let ctx = self.get_context(account_id).await?;
+        let message = Message::load_from_db(&ctx, MsgId::new(instance_msg_id)).await?;
+        Ok(message.get_webxdc_href())
```

Comment:
performance-wise, it'd be better to add "href" as a property to the "js message object", similar to "info type" or "text"

Reply:
Yes that what I also thought would be better, but I wondered if there was a reason to have a explicit function defined

Reply:
I think I will extend it if you agree

Reply:
internally, it is in "param", therefore there is a getter-function, so that UI do not need to think about that detail.

but these getters are cheap, no db-call needed. 

in C-FFI, this is quite strict, as a rule of thumb, i.e. you load a dc_msg_t which requires a single db call - all dc_msg_getters do not use db calls -- in js land, the object loading is similar, but often needs more db calls)

Reply:
yeah, feel free to extend or to keep the current state. this mainly affects desktop and is also easily changable

Reply:
I think in this case explicit function might be better, because the info object might be cached in ui, unless it was updated by message changed event?
Atleast such a dedicated function might be simpler to understand?

Reply:
Since adding the href to the message object is very "cheap", I added it for direct use, where it applies

Reply:
why not in the webxdc_info object?

Reply:
webxdc_info is refers to webxdc-instance, not to a in-chat-status-message (the "info" wording somehow used by both is confusing, but well :)

Reply:
ah of course that makes sense, I was a bit confused because I somehow mixed it in in my head with the "save what info message/notification was clicked and then remember it when the user starts the app".
😆 😅

Reply:
I was also thinking about that. Imho it would make sense to deliver the webxdcInfo also with the WebxdcInfoMessage since the info is anyway needed for handling the click on a WebdcInfoMessage. I'll show my proposal in another PR. https://github.com/deltachat/deltachat-core-rust/pull/6282

And then we can thinking about adding the href as a (optional) property of the WebxdcInfo. But I don't want to mix up too many things.